### ISSUE TITLE
[Bug]: Livewire\CorruptComponentPayloadException on any action when $sortField and Detail Row is set

### ISSUE THREAD

**Issue by P4ulBr4ndt** [Power-Components/livewire-powergrid#712]
### Have you searched through other issues to see if your problem is already reported or has been fixed?

Yes, I did not find it.

### Did you read the documentation?

Yes, I did not find it.

### Have you tried to publish the views?

Yes - I didn't work.

### Is there an error in the console?

![powergrid](https://user-images.githubusercontent.com/79166045/200197718-2833a056-5495-447d-8e49-ce1e82466bd3.png)

### PowerGrid

3.5.10

### Laravel

9.38.0

### Livewire

2.10.7

### Alpine JS

3.10.5

### Theme

Tailwind 3.x with tailwind/forms

### Describe the bug.

I am expecting the Detail Row to open upon clicking the little icon on the left side of the table. The Table is rendering fine and all features are working as expected. However clicking the Detail Row does not open it but trigger the Livewire\CorruptComponentPayloadException in a very minimal setup. My Table does not use a lot of fancy features or data.

### To Reproduce...

Not really much is happending here. An eloquent Model and a powergrid for it.
The primaryKey is not called "id", the query as datasource is very simple. Pretty much clueless why this is happening.

### Extra information

```markdown
class Article extends Model
{
    use HasFactory;

    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'vStandardArtikel';


    /**
     * The primary key associated with the table.
     *
     * @var string
     */
    protected $primaryKey = 'kArtikel';
}

<?php

namespace App\Http\Livewire;

use App\Models\Article;
use Illuminate\Support\Carbon;
use Illuminate\Database\QueryException;
use Illuminate\Database\Eloquent\Builder;
use PowerComponents\LivewirePowerGrid\Rules\{Rule, RuleActions};
use PowerComponents\LivewirePowerGrid\Traits\ActionButton;
use PowerComponents\LivewirePowerGrid\{Button,
    Column,
    Detail,
    Exportable,
    Footer,
    Header,
    PowerGrid,
    PowerGridComponent,
    PowerGridEloquent};

final class ArticleTable extends PowerGridComponent
{
    use ActionButton;

    public string $primaryKey = 'kArtikel';
    public string $sortField = 'cArtNr';

    /*
    |--------------------------------------------------------------------------
    |  Features Setup
    |--------------------------------------------------------------------------
    | Setup Table's general features
    |
    */
    public function setUp(): array
    {
        // $this->showCheckBox();

        return [
            Exportable::make('export')
                ->striped()
                ->type(Exportable::TYPE_XLS, Exportable::TYPE_CSV),
            Header::make()
                ->showSearchInput()
                ->showToggleColumns(),
            Footer::make()
                ->showPerPage()
                ->showRecordCount(),
            Detail::make()
                ->view('components.details')
                ->showCollapseIcon(),
        ];
    }

    /*
    |--------------------------------------------------------------------------
    |  Datasource
    |--------------------------------------------------------------------------
    | Provides data to your Table using a Model or Collection
    |
    */

    /**
    * PowerGrid datasource.
    *
    * @return Builder<Article>
    */
    public function datasource(): Builder
    {
        return Article::query()->where('kWarengruppe', '=', 30);
    }

    /*
    |--------------------------------------------------------------------------
    |  Relationship Search
    |--------------------------------------------------------------------------
    | Configure here relationships to be used by the Search and Table Filters.
    |
    */

    /**
     * Relationship search.
     *
     * @return array<string, array<int, string>>
     */
    public function relationSearch(): array
    {
        return [];
    }

    /*
    |--------------------------------------------------------------------------
    |  Add Column
    |--------------------------------------------------------------------------
    | Make Datasource fields available to be used as columns.
    | You can pass a closure to transform/modify the data.
    |
    | ❗ IMPORTANT: When using closures, you must escape any value coming from
    |    the database using the `e()` Laravel Helper function.
    |
    */
    public function addColumns(): PowerGridEloquent
    {
        return PowerGrid::eloquent()
            ->addColumn('kArtikel')
            ->addColumn('Artikelnummer')
            ->addColumn('Artikelname')
            ->addColumn('EAN')
            ->addColumn('ASIN')
            ->addColumn('AmazonFNSKU');
    }

    /*
    |--------------------------------------------------------------------------
    |  Include Columns
    |--------------------------------------------------------------------------
    | Include the columns added columns, making them visible on the Table.
    | Each column can be configured with properties, filters, actions...
    |
    */

     /**
     * PowerGrid Columns.
     *
     * @return array<int, Column>
     */
    public function columns(): array
    {
        return [
            Column::make('kArtikel', 'kArtikel')
                ->searchable()
                ->makeInputRange('kArtikel')
                ->sortable(),

            Column::make('Artikelnummer', 'cArtNr')
                ->searchable()
                ->makeInputText('cArtNr')
                ->sortable(),

            Column::make('Artikelname', 'cName')
                ->searchable()
                ->makeInputText('cName')
                ->sortable(),

            Column::make('EAN', 'cBarcode')
                ->searchable()
                ->makeInputText('cBarcode')
                ->sortable(),

            Column::make('ASIN', 'cASIN')
                ->searchable()
                ->makeInputText('cASIN')
                ->sortable(),

            Column::make('AmazonFNSKU', 'cAmazonFNSKU')
                ->searchable()
                ->makeInputText('cAmazonFNSKU')
                ->sortable(),
        ];
    }
}


// details.blade.php: As simple as
<div class="p-2 bg-white border border-slate-200">
   test
</div>

```

**Comment by luanfreitasdev** [comment]
The issue is resolved when you remove the property. $sortField. can you test it please?

**Comment by P4ulBr4ndt** [comment]
Removing the property $sortField produces an invalid query as it tries to order by a non existing column id.
select top 10 * from [vStandardArtikel] where [kWarengruppe] = 30 order by [id] asc

But I can confirm it works with another table where no sortField is set as id is available.

Am I not supposed to set sortField like that?

**Comment by P4ulBr4ndt** [comment]
By the way, I debugged to the what part of the component got corrupted. It seems like the primary keys values are kind of shuffled which corrupts the component: See screenshot for state before triggering a Detail Row and after.
![powergrid 2](https://user-images.githubusercontent.com/79166045/200199112-8596d197-f173-40cb-a449-7da8256d9ee0.png)

**Comment by luanfreitasdev** [comment]
It really is strange behavior when adding softField or sortDirection. I will continue investigating. If you find something out, feel free to open a PR 😀

**Comment by P4ulBr4ndt** [comment]
Besides the screenshot showing the actual corruption I have one more observation to share:
Setting sortField to "kArtikel" or "cBarcode" fixes the problem in this case. "cArtNr", "cASIN" and "cName" trigger the crash.
However if the user sorts by this column after I set sortField to an unproblematic value it works fine. 
Very strange behavior indeed. Thanks for your support and this library. Really enjoying working with it.
I will share further information / create a PR if I found something out.

**Comment by P4ulBr4ndt** [comment]
Bug is only triggered if sortField is set and a Detail Row is defined. Bug is then not only triggered by opening the Detail Row, but also by changing page, changing page size, sorting... pretty much any action.

Knowing the Detail Row is the problem and looking at the actual corruption as seen in my screenshot I found the error to be triggered by PowerGridComponent::resolveDetailRow().
https://github.com/Power-Components/livewire-powergrid/blob/1697f04aa30366b1b6c332ccc7e7f1c458faee79/src/PowerGridComponent.php#L325

data_set corrupts the detail.state by reordering it against the order it was set by sortField earlier, which is the reason why it's only happening with sortField != primaryKey. 

Commenting out L322-326  works around this - Detail not working then of course.

@luanfreitasdev Pinning it down to this point wasn't really a problem. Can you think of a way to fix this? Im not sure how to change the logic at that point.

**Comment by luanfreitasdev** [comment]
Thanks for the observation!. I'll look into other ways to get around the problem.

**Comment by luanfreitasdev** [comment]
Fixed in 3.5.11. Thank you for your help

https://github.com/Power-Components/livewire-powergrid/releases/tag/v3.5.11
